### PR TITLE
Change provider configuration keys for OAuth

### LINKED ISSUE #9662: Replace flask_oauthlib with Authlib
**Description**

flask_oauthlib has been deprecated in favour of Authlib. It would be good if airflow starts using Authlib

**Use case / motivation**

FlaskAppBuilder is now using Authlib. 
Since FlaskAppBuilder is deeply integrated into Airflow, it will be good to also have this Authlib. Flask-oauthlib documentation recommends Authlib

**Related Issues**

### PR DESCRIPTION
**Pull request by ephraimbuddy**
---
Closes #9662 
This PR replaces flask-oauthlib configuration keys with Authlib configuration keys as per flask-appbuilder.
See  https://github.com/dpgaspar/Flask-AppBuilder#change-log

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: UPDATING.md [hunk 1]
```diff
@@ -61,6 +61,20 @@ More tips can be found in the guide:
...
 -->
+### Flask-ouathlib has been replaced with Authlib
+The flask-ouathlib has been replaced with authlib because flask-outhlib have also
+been deprecated in favour of authlib.
+The Old and New provider configuration keys that have changed are as follows
+
+|      Old Keys      |      New keys     |
+|--------------------|-------------------|
+| consumer_key       | client_id         |
+| consumer_secrete   | client_secrete    |
```
Comment (ephraimbuddy) [inline UPDATING.md]:
```suggestion
| consumer_secret   | client_secret    |
```

File: setup.py [hunk 2]
```diff
@@ -251,7 +251,7 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
     'facebook-business>=6.0.2',
 ]
 flask_oauth = [
-    'Flask-OAuthlib>=0.9.1',
+    "Authlib>=0.14.3",
     'oauthlib!=2.0.3,!=2.0.4,!=2.0.5,<3.0.0,>=1.1.2',
     'requests-oauthlib==1.1.0',
```
Comment (ephraimbuddy) [inline setup.py]:
```suggestion
   
```

File: UPDATING.md [hunk 3]
```diff
@@ -61,6 +61,20 @@ More tips can be found in the guide:
...
+The flask-ouathlib has been replaced with authlib because flask-outhlib have also
+been deprecated in favour of authlib.
+The Old and New provider configuration keys that have changed are as follows
+
+|      Old Keys      |      New keys     |
+|--------------------|-------------------|
+| consumer_key       | client_id         |
+| consumer_secret   | client_secret    |
+| base_url           | api_base_url      |
+| request_token_url  | client_kwargs     |
```
Comment (ephraimbuddy) [inline UPDATING.md]:
```suggestion
| request_token_params | client_kwargs     |
```

**Comment by ephraimbuddy** [comment]
Hi @mik-laj , can I ask for review?

File: UPDATING.md [hunk 4]
```diff
@@ -61,6 +61,20 @@ More tips can be found in the guide:
...
+The Old and New provider configuration keys that have changed are as follows
+
+|      Old Keys       |      New keys     |
+|---------------------|-------------------|
+| consumer_key        | client_id         |
+| consumer_secret     | client_secret     |
+| base_url            | api_base_url      |
+| request_token_params| client_kwargs     |
+
+For more information, visit https://docs.authlib.org/en/stable/
```
Comment (ephraimbuddy) [inline UPDATING.md]:
```suggestion
For more information, visit https://flask-appbuilder.readthedocs.io/en/latest/security.html
```

**Comment by ephraimbuddy** [comment]
Hi @mik-laj, How do I remove Flask-OAuthlib from the requirement files. I have removed it from setup.py and generated requirements, yet it's still in requirement files.

**Comment by mik-laj** [comment]
Can you run [pipdeptree](https://pypi.org/project/pipdeptree/) and give me output?

**Comment by ephraimbuddy** [comment]
Result of running `pipdeptree -p Flask-OAuthlib`
```
root@40305431169e:/opt/airflow# pipdeptree -p Flask-OAuthlib
Warning!!! Possibly conflicting dependencies found:
* tableauserverclient==0.9
 - urllib3 [required: ==1.24.3, installed: 1.25.9]
* snowflake-connector-python==2.2.8
 - idna [required: <2.10, installed: 2.10]
 - requests [required: <2.24.0, installed: 2.24.0]
 - azure-storage-blob [required: >=12.0.0,<13.0.0, installed: 2.1.0]
 - boto3 [required: >=1.4.4,<1.14, installed: 1.14.20]
* moto==1.3.14
 - idna [required: >=2.5,<2.9, installed: 2.10]
* Flask-OAuthlib==0.9.5
 - oauthlib [required: >=1.1.2,<3.0.0,!=2.0.5,!=2.0.4,!=2.0.3, installed: 3.1.0]
* distributed==2.19.0
 - tornado [required: >=6.0.3, installed: 5.1.1]
* botocore==1.17.20
 - docutils [required: >=0.10,<0.16, installed: 0.16]
* astroid==2.4.2
 - lazy-object-proxy [required: ==1.4.*, installed: 1.5.0]
------------------------------------------------------------------------
Flask-OAuthlib==0.9.5
  - Flask [required: Any, installed: 1.1.2]
    - click [required: >=5.1, installed: 6.7]
    - itsdangerous [required: >=0.24, installed: 1.1.0]
    - Jinja2 [required: >=2.10.1, installed: 2.10.3]
      - MarkupSafe [required: >=0.23, installed: 1.1.1]
    - Werkzeug [required: >=0.15, installed: 0.16.1]
  - oauthlib [required: >=1.1.2,<3.0.0,!=2.0.5,!=2.0.4,!=2.0.3, installed: 3.1.0]
  - requests-oauthlib [required: >=0.6.2, installed: 1.3.0]
    - oauthlib [required: >=3.0.0, installed: 3.1.0]
    - requests [required: >=2.0.0, installed: 2.24.0]
      - certifi [required: >=2017.4.17, installed: 2020.6.20]
      - chardet [required: >=3.0.2,<4, installed: 3.0.4]
      - idna [required: >=2.5,<3, installed: 2.10]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.9]
```

**Comment by kaxil** [comment]
> Hi @mik-laj, How can I remove Flask-OAuthlib from the requirement files. I have removed it from setup.py and generated requirements, yet it's still in requirement files.

It might be required by any other dependency

**Comment by kaxil** [comment]
increase the epoch https://github.com/apache/airflow/blob/master/Dockerfile.ci#L44 and rebuild it show hopefully disappear then

**Comment by ephraimbuddy** [comment]
I have not been able to see the package that is adding `Flask-OAuthlib` but what I realized after trying several times was that `pandas-gbq` has dependency `google-auth-oauthlib` which in turn requires `requests-oauthlib` which depends on `oauthlib`.
`google-cloud-bigquery` also depends on some of the above.

What I am thinking is to remove the version constraints on `oauthlib` and `requests-oauthlib` since we no longer require Flask-OAuthlib because FlaskAppBuilder no longer use it as can be seen here https://github.com/dpgaspar/Flask-AppBuilder#change-log.

It's a breaking change in FlaskAppBuilder and removing the constraints will allow other libraries that requires `requests-oauthlib` and `oauthlib` to use the latest versions.

What's the best way to handle this? @mik-laj @kaxil

**Comment by ephraimbuddy** [comment]
I think since both Flask-Oauthlib and Authlib can coexist, It's safe to only update the breaking change. I have rolled back my changes on setup.py. We already have Authlib in the requirements files, this change now focused on Flask App Builder breaking change

**Review by mik-laj — APPROVED** [review]

**Comment by mik-laj** [comment]
LGTM, but it require rebase.

**Pull request merged**